src/main/java/homework_6/map_problems_collision_generator/MapProblemsMutableGenerator.java

```diff
@@ -0,0 +1,35 @@
+package homework_6.map_problems_collision_generator;
+
+import java.util.Objects;
+import java.util.Random;
+
+public class MapProblemsMutableGenerator {
+
+    private String val;
+
+    public MapProblemsMutableGenerator() {
+        setVal("кладем?");
+    }
+
+    public String getVal() {
+        return val;
+    }
+
+    public void setVal(String val) {
+        this.val = val;
+    }
+
+    @Override
+    public int hashCode() {
+        return new Random().nextInt(Integer.MAX_VALUE - 1);
```

Comment:
Well, there;s no mutable key problem, it's just invalid hashcode implementation. Mutable key generates problem even if hashcode impl is valid

Reply:
Vsevolod corrected hashCode() and seems like mutable problem is appeared!